### PR TITLE
Completed databases, fixed user entities, and other small changes

### PR DESCRIPTION
**Pull request by rafaelgacesa**
General completion of playlist and user databases and database-related entity classes.

### In this pull:

**Entities:**
- finished the `RegularUser` class with a `playlistList` and everything else required
- added some constructors and a `songs` attribute to `artistUser`
- added corresponding required methods to `userFactory`

**Database related classes:**
- updated `playlistLibrary` to save properly so playlists can be updated
   - will be useful for @NicholasMacasaet
   - also did a general refactor of the file I/O methods to simplify them
   - updated `playlistDsData` to deal with privacy changes
- completed the `userList` database since the entities are now complete
   - pretty straightforward, created tests for it
   - updated `userDsData` to create the `User` objects and build their `.csv` output in a `toString()` method
- added some small changes to @chrstinalin 's `songLibrary` to allow tests to pass
   - these changes are all pretty temporary, marked with `TODO`s
   - some changes also made to `songDsData` in a similar vein

### CODE REVIEW

**Review by NicholasMacasaet — APPROVED** [review]
Everything looks straightforward enough. The implementation for the RegularUser things that are necessary for my use case look great. If I have any questions about the database that are relevant to my code I will make sure to ask you. But other than that good work Raf.

File: src/main/java/Database/playlistLibrary.java [hunk 1]
```diff
@@ -77,13 +77,11 @@ private void createFile(String fp){
      */
```
Comment (NicholasMacasaet) [inline src/main/java/Database/playlistLibrary.java:77]: Small change, maybe add some Javadocs here/descriptions

**Review by NicholasMacasaet — CHANGES_REQUESTED** [review]
Small change, maybe add some javadocs for the saveFile method

File: src/main/java/Database/playlistLibrary.java (hunk 1 above)
Comment (rafaelgacesa) [reply src/main/java/Database/playlistLibrary.java:77]: I can't tell which method you're highlighting here. Unless you're asking for some more comments within the methods on logic (which I definitely should add eventually lol), I don't think there's any methods missing javadocs?

Comment (NicholasMacasaet) [reply src/main/java/Database/playlistLibrary.java:77]: Ah sorry this is the saveFile() method, but just adding a small description would be good.

**Review by NicholasMacasaet — APPROVED** [review]

File: src/main/java/Database/songLibrary.java [hunk 2]
```diff
@@ -9,9 +9,10 @@
  */
 public class songLibrary implements songAccessInterface{
 
+    // TODO: change "" param to filepath
     private static final songLibrary SONG_LIBRARY = new songLibrary("");
     private final HashMap<Integer, songDsData> library;
-    private String filepath;
+    private final String filepath;
```
Comment (chrstinalin) [inline src/main/java/Database/songLibrary.java:15]: Whoops

File: src/main/java/Database/songLibrary.java [hunk 3]
```diff
@@ -9,9 +9,10 @@
  */
 public class songLibrary implements songAccessInterface{
 
+    // TODO: change "" param to filepath
```
Comment (chrstinalin) [inline src/main/java/Database/songLibrary.java:12]: Ah yeah my local file's had it changed for a while. I still need to make a pull, but I wanted to finish things up.

**Review by chrstinalin — APPROVED** [review]
Looks good!

File: src/main/java/Database/songLibrary.java (hunk 2 above)
Comment (rafaelgacesa) [reply src/main/java/Database/songLibrary.java:15]: Lol, I was just clicking on all the random warnings when I pulled.

File: src/main/java/Database/playlistDsData.java [hunk 4]
```diff
@@ -21,12 +22,14 @@ public playlistDsData(Playlist p){
         this.playlist = p;
         library = songLibrary.getInstance();
         users = userList.getInstance();
+        this.privacy = 1;
```
Comment (sarxhlwalker) [inline src/main/java/Database/playlistDsData.java:25]: I'd suggest adding a comment here to signify what 1 (vs. 0 presumably) signifies wrt privacy (also I thought we were getting rid of playlist privacy controls? is that still the case?

File: src/main/java/Database/playlistLibrary.java [hunk 5]
```diff
@@ -46,9 +45,10 @@ private Map<Integer, playlistDsData> loadFile() {
                 // assumes 5 columns, properly filled
                 String line = in.nextLine();
                 String[] data = line.split(",");
-                int id = Integer.parseInt(data[0]);
-                saved.add(id);
-                p.put(Integer.parseInt(data[0]), new playlistDsData(data));
+                if (data.length == 5) {
+                    int id = Integer.parseInt(data[0]);
+                    p.put(id, new playlistDsData(data));
+                }
```
Comment (sarxhlwalker) [inline src/main/java/Database/playlistLibrary.java:51]: Should there be some kind of warning of result if the 5 columns are not properly filled and you don't complete loading the file?

**Review by sarxhlwalker — APPROVED** [review]
Overall looks good. Some comments.

File: src/main/java/Database/playlistLibrary.java (hunk 5 above)
Comment (rafaelgacesa) [reply src/main/java/Database/playlistLibrary.java:51]: It just skips the line without 5 columns and moves on, so it should be fine. Theoretically, my database file is the only thing that should be touching the csv files so I think it's an ok assumption.

File: src/main/java/Database/playlistDsData.java (hunk 4 above)
Comment (rafaelgacesa) [reply src/main/java/Database/playlistDsData.java:25]: Yeah, that is gone from the use cases but I don't want to waste time removing it from the databases until we're totally sure it needs to go.

**Pull request merged**